### DISCUSSION TITLE
Image Pull Degredation

### DISCUSSION THREAD

**Discussion by capps-b** [goharbor/harbor#18974] [Q&A]
Hi, 

We are running a harbor cluster managed by helm. The first few pulls are fine, but after several pulls the speed becomes extremely slow. I've also noticed that registry-photon pods are consistently growing in memory usage regardless of traffic and never seems to let that memory go (unless the pod is killed/restarted).
1. What is it cacheing?
2. Is there some memory leak going on? Is it something we misconfigured that is common?
3. Why does it hold onto memory after connections have (presumably) closed?

I want to better understand what the registry pod actually does and how it communicates with the rest of them. Could it be that connections are not getting closed out after the pull is done?

**Comment by chlins** [comment]
Track this by issue https://github.com/goharbor/harbor/issues/18989.
